src/tests/unit/tests/reports/components/report-sections/link-script-provider.test.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { isFunction } from 'lodash';
+import {
+    addEventListenerForLink,
+    getAddListenerForLink,
+} from 'reports/components/report-sections/link-script-provider';
+import { It, Mock, MockBehavior, Times } from 'typemoq';
+
+describe('LinkScriptProvider', () => {
+    describe('getAddListenerForLink', () => {
+        it('matches content', () => {
+            const code = 'this is test code';
+
+            const result = getAddListenerForLink(code);
+
+            expect(result).toMatchSnapshot();
+        });
+    });
+
+    describe('addEventListenerForLink', () => {
+        it('adds listener for click event on target-page-link', () => {
+            const targetPageLinkMock = Mock.ofType<HTMLElement>();
+            const docMock = Mock.ofType<Document>(undefined, MockBehavior.Strict);
+            docMock
+                .setup(doc => doc.getElementById('target-page-link'))
+                .returns(() => targetPageLinkMock.object);
+
+            addEventListenerForLink(docMock.object);
+
+            targetPageLinkMock.verify(
+                link => link.addEventListener('click', It.is(isFunction)),
```

Comment:
This `verify` probes we are adding and event listener to the proper element on the dom.
We are missing a test when we actually click that element, triggering the listener function. 
Doing so will likely make you change how we are using `confirm` as mention on the comment I left on the actual usage of `confirm`